src/devices/llama_devices.ts

```diff
@@ -0,0 +1,150 @@
+    values: [
+        {
+            name: 'on/off',
+            value: 0,
+        },
+        {
+            name: 'off/on',
+            value: 1,
+        },
+        {
+            name: 'toggle',
+            value: 2,
+        },
+    ],
+};
+const configs: Config[] = [
+    switchTypeConf,
+    switchAtionConf,
+];
+
+function mapModernEnum(cfg: Config, epName: string) {
+    return modernExtend.enumLookup({
+        name: cfg.name,
+        lookup: mapObject(cfg.values, (k) => k.name, (v) => v.value),
+        endpointName: epName,
+        cluster: 'genOnOffSwitchCfg',
+        attribute: cfg.attribute,
+        description: cfg.description,
+        access: 'ALL',
+        entityCategory: 'config',
+    });
+}
+
+function addEndpointPrefix(ep: number) {
+    return `${prefixEnpointName}${ep}`;
+}
+
+function createSwitchExpends(epc: number) {
+    const features: ModernExtend[] = [];
+    for (let ep = 1; ep <= epc; ep++) {
+        for (let i = 0; i < configs.length; i++) {
+            features.push(mapModernEnum(configs[i], addEndpointPrefix(ep)));
+        }
+    }
+    return features;
+}
+
+function mapObject<T>(arr: T[], mapKey: (k: T) => string | [string], mapValue: (v: T) => string | number) {
+    return Object.fromEntries(arr.map((el, i, arr) => [mapKey(el), mapValue(el)]));
+}
+
+const fzLocal = {
+    on_off_action: {
+        cluster: 'genOnOff',
+        type: [
+            'commandOn',
+            'commandOff',
+            'commandToggle',
+            // 'commandOnWithTimedOff'
+        ],
+        convert: (model, msg, publish, options, meta) => {
+            return {action: `${msg.endpoint.ID}_${msg.type.replace('command', '')}`};
+        },
+    } satisfies Fz.Converter,
+};
+
+const definitions: Definition[] = [
+    {
+        zigbeeModel: ['Zucchini'],
+        model: 'Zucchini',
+        vendor: 'Llama Devices',
+        description: 'On/Off output device',
+        fromZigbee: [fzLocal.on_off_action],
+        exposes: [
+            e.action([]),
+        ],
+        extend: [
+            modernExtend.battery({
+                percentageReporting: true,
+                voltageReporting: false,
+                percentageReportingConfig: {change: 0, max: 3600, min: 0},
+            }),
+            modernExtend.temperature({
+                reporting: {min: 0, max: 3600, change: 0},
+            }),
+            modernExtend.humidity({
+                reporting: {min: 0, max: 3600, change: 0},
+            }),
+            ...createSwitchExpends(endpointCount),
+        ],
+        configure: async (device, coordinatorEndpoint, logger) => {
+            device.endpoints.filter((ep) => ep.supportsInputCluster('genOnOffSwitchCfg'))
+                .forEach(async (ep) => {
+                    await ep.read('genOnOffSwitchCfg', configs.map((c) => c.attribute));
+                });
+            device.save();
+        },
+        endpoint: (device) => {
```

Comment:
Use the `deviceEndpoints()` modern extend instead of this.

Reply:
deviceEndpoints() modern extend does not provide device instance. i have to hardcode endpoints?

Reply:
I already hardcoded enpointCount and it hearts... :(

Reply:
done